packages/shared_preferences/lib/shared_preferences.dart

```diff
@@ -152,6 +152,15 @@ class SharedPreferences {
     return await _kChannel.invokeMethod<bool>('clear');
   }
 
+  /// Set App Domain / Group
+  /// This method only works for iOS and does nothing in Android
+  Future<void> setDomain(String domain) async {
```

Comment:
If this only works on iOS, should we throw when using it in Android with `throw UnsupportedError('Functionality not available on Android');`? (just like other plugins do).

```suggestion
  Future<void> setDomain(String domain) async {
    if (Platform.isAndroid) {
      throw UnsupportedError('Functionality not available on Android');
    }
    final Map<String, dynamic> params = <String, dynamic>{
      'domain': domain,
    };
    _kChannel.invokeMethod<void>('setDomain', params);
  }
```

Reply:
I wouldn't do that because its not really an error, it just gets ignored by Android but you can still use setDomain without need to check if you're on iOS.